src/tty_util.rs

```diff
@@ -0,0 +1,30 @@
+use libc::{c_int, c_ulong, pid_t, sighandler_t};
+use libc::signal;
+use nix::sys::ioctl::ioctl;
+use nix::sys::signal::{SIGTTIN, SIGTTOU};
+
+
+static SIG_IGN: sighandler_t = 1;
+static SIG_ERR: sighandler_t = !0;
+
+static TIOCSPGRP: c_ulong = 21520;
+
+
+pub fn give_tty_to(fd: c_int, pid: pid_t) -> Result<(), String> {
+    let res = unsafe { ioctl(fd, TIOCSPGRP, &pid) };
+    match res {
+        res if res < 0 => Err(format!("Error when giving tty with fd {} to process {}", fd, pid)),
+        _ => Ok(()),
+    }
+}
+
+pub fn ignore_tty_signals() -> Result<(), String> {
+    let tty_signals = vec![SIGTTIN, SIGTTOU];
```

Comment:
It's not "rusty" to allocate a vector just to iterate over the items. Should just call `sigaction()` twice. (And yes `sigaction` is a safe wrapper around the same stuff.